lib/ack_tracker.h

```diff
@@ -84,4 +84,19 @@ ack_tracker_manage_msg_ack(AckTracker *self, LogMessage *msg, AckType ack_type)
   self->manage_msg_ack(self, msg, ack_type);
 }
 
+static inline void
+_ack_tracker_update_source_when_ack_received(LogSource *source, AckType ack_type)
+{
+  if (ack_type == AT_SUSPENDED)
+    {
+      if (log_source_free_to_send(source))
```

Comment:
I think we should not check free_to_send() here. We might skip suspend() while another thread wakes up the source.